addons/test_mail/tests/test_mail_channel.py

```diff
@@ -245,6 +245,24 @@ def test_channel_auto_unsubscribe_archived_or_deleted_users(self):
         self.assertEqual(test_channel_group.channel_partner_ids, self.env['res.partner'])
         self.assertEqual(self.test_channel.channel_partner_ids, self.user_employee.partner_id | test_partner)
 
+    def test_multi_company_chat(self):
+        company_A = self.env['res.company'].create({'name': 'Company A'})
+
+        test_user = self.env['res.users'].create({
+            'login': 'nomail',
+            'name': 'No Mail',
+            'company_ids': [(6, 0, company_A.ids)],
+            'company_id': company_A.id
+        })
+        test_chat = self.env['mail.channel'].with_context(self._test_context).create({
+            'channel_partner_ids': [(4, test_user.partner_id.id), (4, self.user_admin.partner_id.id)],
+            'public': 'private',
+            'channel_type': 'chat',
+            'name': 'test'
+        })
+        initial_channel_info = self.env['mail.channel'].with_context(allowed_company_ids=self.user_admin.company_ids.ids).channel_get(test_user.partner_id.ids)
```

Comment:
Instead of using admin which is often a special user, I would create a 2nd normal user and a 2nd company in the test, and open the chat between User A/Company A and User B/Company B